packages/@lwc/synthetic-shadow/src/shared/event-target.ts

```diff
@@ -29,20 +51,17 @@ export function getEventListenerWrapper(fnOrObj: unknown) {
     let wrapperFn = EventListenerMap.get(fnOrObj);
     if (isUndefined(wrapperFn)) {
         wrapperFn = function (this: EventTarget, event: Event) {
+            const currentTarget = eventCurrentTargetGetter.call(event) as EventTarget;
```

Comment:
Instead of using a type guard, you can use `!` operator when checking against a nullable value. I would also recommend adding a comment above explaining that it is safe to consider that `currentTarget` can't be `null` here because this function is invoked from an event listener.

```suggestion
            const currentTarget = eventCurrentTargetGetter.call(event)!;
```